includes/admin/upgrades/upgrade-functions.php

```diff
@@ -556,9 +600,140 @@ function edd_v21_upgrade_customers_db() {
 		// No more customers found, finish up
 
 		update_option( 'edd_version', preg_replace( '/[^0-9.].*/', '', EDD_VERSION ) );
+		delete_option( 'edd_doing_upgrade' );
 
 		wp_redirect( admin_url() ); exit;
 	}
 
 }
 add_action( 'edd_upgrade_customers_db', 'edd_v21_upgrade_customers_db' );
+
+/**
+ * Fixes the edd_log meta for 2.2.6
+ *
+ * @since 2.2.6
+ * @return void
+ */
+function edd_v226_upgrade_payments_db() {
+
+	global $wpdb;
+
+	if( ! current_user_can( 'manage_shop_settings' ) ) {
+		wp_die( __( 'You do not have permission to do shop upgrades', 'edd' ), __( 'Error', 'edd' ), array( 'response' => 403 ) );
+	}
+
+	ignore_user_abort( true );
+
+	if ( ! edd_is_func_disabled( 'set_time_limit' ) && ! ini_get( 'safe_mode' ) ) {
+		@set_time_limit(0);
+	}
+
+	$step   = isset( $_GET['step'] ) ? absint( $_GET['step'] ) : 1;
+	$number = 25;
+	$offset = $step == 1 ? 0 : ( $step - 1 ) * $number;
+
+	if ( 1 === $step ) {
+		// Check if we have any variable price products on the first step
+		$args = array(
+			'post_type'  => 'download',
+			'meta_key'   => '_variable_pricing',
+			'meta_value' => '1',
+			'posts_per_page' => -1
+		);
+
+		$sql = "SELECT ID FROM $wpdb->posts p LEFT JOIN $wpdb->postmeta m ON p.ID = m.post_id WHERE m.meta_key = '_variable_pricing' AND m.meta_value = 1";
+		$has_variable = $wpdb->get_col( $sql );
+
+		if( empty( $has_variable ) ) {
+			// We had no variable priced products, so go ahead and just complete
+			update_option( 'edd_version', preg_replace( '/[^0-9.].*/', '', EDD_VERSION ) );
+
+			wp_redirect( admin_url() ); exit;
+		}
+
+	}
+
+	$payment_ids = $wpdb->get_col( $wpdb->prepare( "SELECT ID FROM $wpdb->posts WHERE post_type = 'edd_payment' ORDER BY post_date DESC LIMIT %d,%d;", $offset, $number ) );
+
+	if( ! empty( $payment_ids ) ) {
+
+		foreach( $payment_ids as $payment_id ) {
+
+			$payment_downloads  = edd_get_payment_meta_downloads( $payment_id );
+			$variable_downloads = array();
+			foreach ( $payment_downloads as $download ) {
+
+				// Don't care if the download is a single price id
+				if ( ! isset( $download['options']['price_id'] ) ) {
+					continue;
+				}
+
+				$variable_downloads[] = array( 'id' => $download['id'], 'price_id' => $download['options']['price_id'] );
+			}
+
+			$variable_download_ids = array_unique( wp_list_pluck( $variable_downloads, 'id' ) );
+			$unique_download_ids   = implode( ',', $variable_download_ids );
+			// Get all Log Ids where the post parent is in the set of download IDs we found in the cart meta
+			$logs = $wpdb->get_results( "SELECT m.post_id AS log_id, p.post_parent AS download_id FROM $wpdb->postmeta m LEFT JOIN $wpdb->posts p ON m.post_id = p.ID WHERE meta_key = '_edd_log_payment_id' AND meta_value = $payment_id AND p.post_parent IN ($unique_download_ids)", ARRAY_A );
+
+			$mapped_logs = array();
+
+			// Go through each cart download
+			foreach( $variable_downloads as $download ) {
```

Comment:
Let's change `$download` to something else so that we don't confuse ourselves with the loop above that also uses `$download`.